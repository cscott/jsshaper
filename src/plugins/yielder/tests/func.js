var log = (typeof console !== "undefined") && console.log || print;
(typeof require !== 'undefined') && require('setup.js');

function foo() {
    log(!!sum);
    function sum(x, y) { return x + y; }
    log(!!sum);
};/* deliberate trailing semicolon */
function foo2(b) {
    log(!!sum);
    if (b) {
        function sum(x, y) { return x + y; }
    }
    log(!!sum);
}
function bar() {
    log(!!sum);
    var sum = function sum2(x, y) { return x + y; };
    log(!!sum);
}
var x = 2;
function baz() {
    log(x);
    var x = 3;
    log(x);
}
var e = 42;
function bat() {
    try {
        throw 1;
    } catch (e) {
        log(e);
        function xx() { return e; }
        var x = function x2() { return e; };
        log(x());
        log(xx());
    }
}
foo();
log('vs');
foo2(false);
log('vs');
foo2(true);
log('vs');
bar();
log('---');
baz();
log('---');
bat();
