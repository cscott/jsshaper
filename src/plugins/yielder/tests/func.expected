true
true
vs
false
false
vs
false
true
vs
false
true
---
undefined
3
---
1
1
1
